docs/how-to-guides/azure_resource_provision.json

```diff
@@ -2,47 +2,66 @@
     "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
     "contentVersion": "1.0.0.0",
     "parameters": {
-
-
         "resourcePrefix": {
-            "type": "string",
+            "minLength": 3,
+            "maxLength": 15,
+            "type": "String",
             "metadata": {
                 "description": "Resource prefix for all the resource provisioned. This should be an alphanumeric string."
-            },
-            "minLength": 3,
-            "maxLength": 15
+            }
         },
         "principalId": {
-            "type": "string",
+            "type": "String",
             "metadata": {
                 "description": "Specifies the principal ID assigned to the role. You can find it by logging into 'https://shell.azure.com/bash' and run 'az ad signed-in-user show --query objectId -o tsv'"
             }
         },
         "allowAllConnections": {
-            "type": "string",
-            "allowedValues": [ "true", "false" ],
             "defaultValue": "true",
+            "allowedValues": [
+                "true",
+                "false"
+            ],
+            "type": "String",
             "metadata": {
                 "description": "Specifies whether to allow client IPs to connect to Synapse"
             }
         },
         "provisionPurview": {
-            "type": "string",
-            "allowedValues": [ "true", "false" ],
             "defaultValue": "true",
+            "allowedValues": [
+                "true",
+                "false"
+            ],
+            "type": "String",
             "metadata": {
                 "description": "Whether or not put purview in the provision script"
             }
+        },
+        "provisionEventHub": {
+            "defaultValue": "true",
+            "allowedValues": [
+                "true",
+                "false"
+            ],
+            "type": "String",
+            "metadata": {
+                "description": "Whether or not to deploy eventhub provision script"
+            }
         }
     },
-    "functions": [],
     "variables": {
         "location": "[resourceGroup().location]",
         "tenantId": "[subscription().tenantId]",
         "redisCacheName": "[concat(parameters('resourcePrefix'),'redis' )]",
         "keyVaultName": "[concat(parameters('resourcePrefix'),'kv')]",
+        "eventhubNameSpaceName": "[concat(parameters('resourcePrefix'),'ehnamespace')]",
```

Comment:
make the name shorter to avoid name length limitation?

Reply:
Fixed